### ISSUE TITLE
 the det_masks is all 0 after utils.unmold_mask function,and AP is 0 

### ISSUE THREAD

**Issue by bing1zhi2** [matterport/Mask_RCNN#1585]
I train a model and try to  run test on it ,the model seems good because the bounding boxs have high confidence. But I found  the AP is zero.  so I run  "Predicted Masks" code to see the mask:
`# Get predictions of mask head
mrcnn = model.run_graph([original_image], [
    ("detections", model.keras_model.get_layer("mrcnn_detection").output),
    ("masks", model.keras_model.get_layer("mrcnn_mask").output),
])`
the output is :
detections               shape: (1, 100, 6)           min:    0.00000  max:    6.00000  float32
masks                    shape: (1, 100, 28, 28, 9)   min:    0.00000  max:    0.99173  float32

It seems no problem,But When I do next step:
`# Masks
det_boxes = utils.denorm_boxes(mrcnn["detections"][0, :, :4], original_image.shape[:2])
det_mask_specific = np.array([mrcnn["masks"][0, i, :, :, c] 
                              for i, c in enumerate(det_class_ids)])
det_masks = np.array([utils.unmold_mask(m, det_boxes[i], original_image.shape)
                      for i, m in enumerate(det_mask_specific)])
log("det_mask_specific", det_mask_specific)
log("det_masks", det_masks)`

the output:

det_mask_specific        shape: (6, 28, 28)           min:    0.00000  max:    0.51804  float32
det_masks                shape: (6, 1024, 1024)       min:    0.00000  max:    0.00000  bool

the det_masks is all 0,I think this is the reason why AP is 0 . Maybe the number in the mask all smaller than the threshold in unmold_mask function calls the problem , but the max number in det_mask_specific  is  0.51804   ,bigger than the     threshold  which is 0.5, so I don't konw what's worng.   I just  train a few epoch ,and II only have few images ,maybe mrcnn_mask_loss is still  high?

**Comment by bing1zhi2** [comment]
I try to train with bigger bitch_size and more epochs ,there is no problem.I turns out that I didn't train the model long enough.
